### PR TITLE
Feature/curl

### LINKED ISSUE #874: Feature Request: libcurl
Being able to call HTTP(S) web servers is a must in the modern day. Though this can be somewhat emulated with sockets, SSL isn't supported and it can be nasty to get right. Would you consider adding libcurl for supporting calling HTTPS endpoints from inside the MUD?

### PR DESCRIPTION
**Pull request by justinm**
Closes #874 by adding libcurl support for calling web services via new efuns.

### CODE REVIEW

**Comment by thefallentree** [comment]
please update CI scripts to install libcurl-dev on various platform

**Comment by thefallentree** [comment]
I can help fix windows/osx build, but can you also add some small LPC tests in testsuite/single/tests/efuns/ ?

File: src/packages/curl/curl.cc [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+#endif
+
+#ifdef F_CURL_POST
+void f_curl_post(void) {
+  LPC_STRING url;
+  LPC_STRING data;
+  mapping_t *headers = nullptr;
+  int num_arg = st_num_arg;
+
+  CHECK_TYPES(sp, T_STRING, 0, F_CURL_POST);
```
Comment (thefallentree) [inline src/packages/curl/curl.cc:168]: There is no need to check type for non-mixed argument, the driver automatically enforce the declaration, so you can remove this

**Comment by gesslar** [comment]
Since this isn't using callbacks, is it working on the main thread and affect the game's performance while it waits for a response? Or how do we know when it has the response?

**Comment by gesslar** [comment]
Additionally, would it make sense to have a single, generic function, that the developer can pass the method ("GET", "POST", "PUT", "PATCH", "DELETE") as an argument and the results come back from curl, rather than having specific GET and POST functions?

**Comment by kparadine** [comment]
> Since this isn't using callbacks, is it working on the main thread and affect the game's performance while it waits for a response? Or how do we know when it has the response?

It does technically use callbacks but the entire call to libcurl is cleaned up within the same function, so the expectation is that the call completed in the main thread.  I think you have a point, in other words.

**Comment by justinm** [comment]
My apologies for never looping back to this PR. I had started adding this support to my lib but have become too distracted with other things. I hope to get back to this PR soon, but I have no real ETA on doing so. I hope to have at least outlined how CURL support could be added, even if I may not be able to see it through.

**Comment by thefallentree** [comment]
Closed due to inaction, plus in favor of using external commands

**Pull request closed without merging**